Utilities/rANS/include/rANS/LiteralDecoder.h

```diff
@@ -21,42 +21,43 @@
 #include <cstddef>
 #include <type_traits>
 #include <iostream>
+#include <iomanip>
 #include <string>
 
 #include <fairlogger/Logger.h>
 
-#include "internal/DecoderSymbol.h"
-#include "internal/ReverseSymbolLookupTable.h"
-#include "internal/SymbolTable.h"
-#include "internal/Decoder.h"
+#include "rANS/internal/DecoderSymbol.h"
+#include "rANS/internal/ReverseSymbolLookupTable.h"
+#include "rANS/internal/SymbolTable.h"
+#include "rANS/internal/Decoder.h"
+#include "rANS/internal/DecoderBase.h"
 
 namespace o2
 {
 namespace rans
 {
 
 template <typename coder_T, typename stream_T, typename source_T>
-class LiteralDecoder : public Decoder<coder_T, stream_T, source_T>
+class LiteralDecoder : public internal::DecoderBase<coder_T, stream_T, source_T>
```

Comment:
If it is really "internal" I would reccomend `private`.

Reply:
Well the Base classes for the Encoders and Decoders expose public methods that should be used. Currently it's just a mean to make code more maintainable and avoid code duplication but no doubt it's far from an ideal solution. Once I have all features implemented, this will have to be refactored. For now I need the flexibility for my research.